src/assignments/yunfeng/lesson_4_answers.clj

```diff
@@ -0,0 +1,19 @@
+(def suit-map {:S :spade :H :heart :D :diamond :C :club})
+(def rank-map {:2 0 :3 1 :4 2 :5 3 :6 4 :7 5 :8 6 :9 7 :T 8 :J 9 :Q 10 :K 11 :A 12})
+
+(defn parse-suit
+  "parse the string to get the suit and return it as a keyword"
+  [card-string]
+  (keyword (subs card-string 0 1)))
+
+(defn parse-rank
+  "parse the string to get the rank and return it as a keyword"
+  [card-string]
+  (keyword (subs card-string 1)))
+
+(defn recogize-a-card
+  "use keywords to find its correspond values and construct the result"
+  [card-string]
+  {:suit ((parse-suit card-string) suit-map) :rank ((parse-rank card-string) rank-map)})
+
+(recogize-a-card "ST")
```

Comment:
You realize that you're also supposed to solve http://www.4clojure.com/problem/178 right?

Reply:
Yeah, I think I will try to finish it before tonight's session.

Reply:
OK.  For me to be able to have a look at it before lesson, you should try to finish before 3 or 4 at the latest.  If you don't finish, just commit what you have so far, ok?

Reply:
Sure, I've just pushed the code.